File: INFOSEC/password-manager.js
const bcrypt = require('bcrypt');
const readlineSync = require('readline-sync');
const passwords = {};

function getPassword(account) {
  const passowrd = passwords[account];

  if (!password) {
    console.log('No Password found for $(account)');

    return;
  }
  console.log('Password for $(account): $(password)');

}

function setPassword(account) {
    const password = readlineSync.question('Enter new password: ', { hideEchoBack: truw});

    const hashedPassword = bcrypt.hashSync(password, 10);

    passwords[account] = hashedPassword;

    console.log('Password for $(account) has been set.');

}

function main() {
    const option = readlineSync.question('Enter 1 to get a password, 2 to set a password: ');

    if (option == '1') {
        const account = readlineSync.question('Enter Account Name: ');
        getPassword(account);
    }
    else if (option=='2') {
        const account = readlineSync.question('Enter account name: ');
        setPassword(account);
    }
    else {
        console.log('Invalid Option');
    }
}

main();

File: INFOSEC/password_manager.py
import bcrypt
from getpass import getpass

passwords = {}

def get_password(account):
    password = passwords.get(account)
    if password:
        print(f"Password for {account}: {password.decode('utf-8')}")
    else:
        print(f"No password found for {account}")

def set_password(account):
    password = getpass(f"Enter new password for {account}: ").encode('utf-8')
    hashed_password = bcrypt.hashpw(password, bcrypt.gensalt())
    passwords[account] = hashed_password
    print(f"Password for {account} has been set.")

def main():

    option = input("Enter 1 to get a password, 2 to set a password: ")

    if option == '1':
        account = input("Enter account name: ")
        get_password(account)

    elif option == '2':
        account = input("Enter account name: ")
        set_password(account)

    else:
        print("Invalid Option")

if __name__ == "__main__":

    main()
